src/functionalTest/groovy/com/autonomousapps/jvm/JavaToolchainSpec.groovy

```diff
@@ -0,0 +1,36 @@
+package com.autonomousapps.jvm
+
+import com.autonomousapps.jvm.projects.JavaToolchainProject
+
+import static com.autonomousapps.utils.Runner.build
+import static com.google.common.truth.Truth.assertThat
+
+final class JavaToolchainSpec extends AbstractJvmSpec {
+  def "does not fail with Java #javaToolchainVersion toolchain (#gradleVersion)"() {
+    given: "a Gradle project using Java toolchain version #javaToolchainVersion that has been built"
+    def project = new JavaToolchainProject(javaToolchainVersion)
+    gradleProject = project.gradleProject
+    build(gradleVersion, gradleProject.rootDir, 'build')
+
+    when: 'running buildHealth'
+    def result = build(gradleVersion, gradleProject.rootDir, 'buildHealth')
+
+    then: 'abiAnalysisMain and explodeByteCodeSourceMain tasks were ran'
+    // These two are the ones that were actually failing, sanity check that java library plugin is applied
+    assert result.tasks.any { (it.getPath() == ':proj:abiAnalysisMain') }
+    assert result.tasks.any { (it.getPath() == ':proj:explodeByteCodeSourceMain') }
+
+    and: 'the report should be empty'
+    assertThat(project.actualBuildHealth()).containsExactlyElementsIn(project.expectedBuildHealth)
+
+    where:
+    // Comment out all but the last for speed
+    gradleVersion | javaToolchainVersion
+    GRADLE_7_5    | 18
+    GRADLE_7_5    | 19
+    GRADLE_7_6    | 18
+    GRADLE_7_6    | 19
```

Comment:
Questions

1. Do you want this to use the `gradleVersions()` method like the other tests?
2. Do you want me to query Adoptium API to try and get latest version of Java so it's easier to catch? It's gross but it might be useful. Not sure your opinion.
3. If I do multiple versions of Java toolchain and gradle versions, do you want me to make it so it's all toolchains on one version of gradle then all versions of gradle with one version of toolchain? That way it's not n^2 and just 2n tests haha.